torch_geometric/loader/node_loader.py

```diff
@@ -143,8 +154,103 @@ def _get_iterator(self) -> Iterator:
         if self.filter_per_worker:
             return super()._get_iterator()
 
+        # if self.device.type == 'cpu' and not self.cpu_affinity_enabled:
+        # TODO: Add manual page for best CPU practices for PyG and switch on warning message
+        # link = ...
+        # Warning(f'Dataloader CPU affinity opt is not enabled, consider switching it on '
+        #             f'(see enable_cpu_affinity() or CPU best practices for PyG [{link}])')
         # Execute `filter_fn` in the main process:
         return DataLoaderIterator(super()._get_iterator(), self.filter_fn)
 
+    def __enter__(self):
+        return self
+
     def __repr__(self) -> str:
         return f'{self.__class__.__name__}()'
+
+    @contextmanager
+    def enable_cpu_affinity(self, loader_cores: List[int] = None):
+        """ Helper method for enabling cpu affinity for compute threads and dataloader workers
```

Comment:
```suggestion
        r"""A context manager to enable cpu affinity for compute threads and dataloader workers
```

Reply:
Please let me know what you think about such description:
"A context manager to enable CPU affinity for DataLoader workers.
        Only for CPU devices!
        As of now, be default it uses NUMA node 0 cores, for a multi-node system.
        This will be gradually extended to increase performance on dual socket CPUs.
        
        Affinitization places DataLoader workers threads on specific CPU cores. In effect, 
        it allows for more efficient local memory allocation and reduces remote memory calls.
        Every time a process or thread moves from one core to another, registers and caches 
        need to be flushed and reloaded. This can become very costly if it happens often, 
        and our threads may also no longer be close to their data, or be able to share data in a cache.
        
        Important:
        If you want to further affinitize compute threads (i.e. with OMP), please make sure that you exclude 
        loader_cores from the list of cores available for compute. This will cause core oversubsription
        and exacerbate performance.
        Args:
            loader_cores : [int] (optional)
                List of cpu cores to which dataloader workers should affinitize to.
                By default cpu0 is reserved for all auxiliary threads & ops.
                DataLoader wil affinitize to cores starting at cpu1.
                default: node0_cores[1:num_workers]
        Usage:
        with dataloader.enable_cpu_affinity(losfrt):
        <training loop>"